### PR TITLE
Use deployment primitives from travis

### PR DESCRIPTION
**Pull request by guilhem**
Need advice on this.
I may go further to improve building

### CODE REVIEW

**Comment by errm** [comment]
This looks good to me

ping @containous/traefik

**Comment by errm** [comment]
Needs a rebase

**Comment by guilhem** [comment]
@errm id you are talking about multi-commit, maintainers can merge by squashing commits ;)

**Comment by errm** [comment]
No just the branch was stale...

**Comment by emilevauge** [comment]
Ping @containous/traefik, we need a review on this ;)

**Comment by vdemeester** [comment]
@guilhem needs a rebase. How could we test that though ?

**Comment by emilevauge** [comment]
@guilhem are you sure that the behavior is the same in the case of a pull request (`deploy-pr.sh` removed) ?

**Comment by errm** [comment]
Yeah looks like the behaviour from deploy-pr is not replicated, can we add that back in please @guilhem ?

**Comment by guilhem** [comment]
@errm @emilevauge Yes, I didn't replicated `deploy-pr.sh` because it's just simply not working :)

https://travis-ci.org/containous/traefik/jobs/181516876#L1946
https://travis-ci.org/containous/traefik/jobs/181516879#L1949

**Comment by emilevauge** [comment]
@guilhem it is https://travis-ci.org/containous/traefik/jobs/183995719#L2089 ;)

**Comment by guilhem** [comment]
humm...
So I have to fix it ^^

**Comment by guilhem** [comment]
@emilevauge I think I've found the problem.
Only PR done by you (or people with write access) have this feature working because of this: https://docs.travis-ci.com/user/pull-requests#Pull-Requests-and-Security-Restrictions

**Comment by emilevauge** [comment]
@guilhem 
https://github.com/containous/traefik/pull/936
https://travis-ci.org/containous/traefik/jobs/182236245#L2087

**Comment by guilhem** [comment]
@emilevauge but https://travis-ci.org/containous/traefik/jobs/182320106#L2085 fail :)

**Comment by guilhem** [comment]
But maybe we want to have a rolling docker for each commit pushed inside "master"

**Comment by emilevauge** [comment]
Yes, check the merge build, not the pull request build: https://travis-ci.org/containous/traefik/jobs/182236245#L2087

**Comment by guilhem** [comment]
@emilevauge should be ok. Plus:
- speed improved
- right docker version tested

**Comment by emilevauge** [comment]
@guilhem I don't think creating a Docker image by commit is a good idea. Usually, we need to atomically take the whole pull request to get it working. Could you rollback to the old behavior?

**Comment by guilhem** [comment]
@emilevauge docker image is pushed not by commit, but with "latest" commit (the merged if there is one).

**Comment by emilevauge** [comment]
Oops, you right ;)

**Comment by guilhem** [comment]
gentleping @containous/traefik

**Comment by guilhem** [comment]
@vdemeester done :)

**Comment by emilevauge** [comment]
@guilhem do you know why tests are failing ?

**Comment by guilhem** [comment]
@emilevauge no idea. can you try to re-run failing tests?

**Comment by emilevauge** [comment]
@guilhem I relaunch the tests 2 times and still the same result :'(

**Comment by guilhem** [comment]
@emilevauge I think bug come from `github.com/libkermit/compose` or `github.com/docker/libcompose` or association of it.
Test for docker 1.10 are ok but compose is failing with docker 1.9
It's quite hard to update dependencies between all this projects as it's used by `traefik` and `traefik/integration` and they are different packages.

Solutions can be:
* remove docker 1.9 support (it's not really tested with current setup).
* To include integrations test in same sources and use [flag matching](http://peter.bourgon.org/go-in-production/#testing-and-validation)

**Comment by guilhem** [comment]
it works with docker 1.10 and 1.12

**Comment by emilevauge** [comment]
@guilhem we agree with @vdemeester to remove Docker 1.9 support :)

**Comment by guilhem** [comment]
@emilevauge I'm in holiday right now, but I will do this for Monday before meetup ;)

File: .travis.yml [hunk 1]
```diff
@@ -1,32 +1,69 @@
...
+  - make test-unit
+  - make test-integration
+after_failure:
+  - docker ps
+
+matrix:
+  fast_finish: true
+  include:
+  - env: DOCKER_VERSION=1.10.3
+  - env: DOCKER_VERSION=1.12.5
```
Comment (vdemeester) [inline .travis.yml]: Should be `1.12.6` 👼 
We could also add `1.13.0` but let's do that in a follow-up 👼

**Comment by guilhem** [comment]
@vdemeester rebased, squashed, fixed etc
:)

**Comment by guilhem** [comment]
```
FAIL: basic_test.go:76: SimpleSuite.TestPrintHelp
basic_test.go:89:
    c.Assert(string(output), checker.Contains, "Usage:")
... obtained string = ""
... substring string = "Usage:"
```
:(

**Comment by emilevauge** [comment]
@vdemeester, any idea on what's going on here ?

**Comment by vdemeester** [comment]
Hum I don't see in the failed build but it might be a weirdiness with how the test is written — if the run is particularly slow it might fail 😓 . I'll take a look.

**Pull request merged**